src/main/java/com/snowplowanalytics/redash/RedashClient.java

```diff
@@ -0,0 +1,437 @@
+     * @throws IllegalArgumentException if no user or no user group with provided userId and userGroupId accordingly exist
+     *                                  onto Redash server.
+     * @throws IOException              if server is not available due to connection error or API key is invalid.
+     */
+    public boolean addUserToGroup(int userId, int groupId) throws IOException {
+        checkIfEntityExists(User.class, userId);
+        UserGroup userGroup = getWithUsersAndDataSources(groupId);
+        if (userGroup.getUsers().stream().anyMatch(u -> u.getId() == userId)) {
+            return false;
+        }
+        String url = baseUrl + GROUPS_URL_PREFIX + "/" + groupId + MEMBERS_URL_PREFIX + API_KEY_URL_PARAM + apiKey;
+        post(url, new JSONObject().put(USER_ID, userId).toString(), CheckResponseStatus.NO);
+        return true;
+    }
+
+    /**
+     * Adds data-source to user-group which are both specified by their ids in methods arguments.
+     *
+     * @return false if data-source already attached to group, true if not and has just added successfully to group
+     * @throws IllegalArgumentException if data-source or user-group does not exist
+     * @throws IOException              if server is not available due to connection error or API key is invalid.
+     */
+    public boolean addDataSourceToGroup(int dataSourceId, int groupId) throws IOException {
+        checkIfEntityExists(DataSource.class, dataSourceId);
+        UserGroup userGroup = getWithUsersAndDataSources(groupId);
+        if (userGroup.getDataSources().stream().anyMatch(ds -> ds.getId() == dataSourceId)) {
+            return false;
+        }
+        String url = baseUrl + GROUPS_URL_PREFIX + "/" + groupId + DATA_SOURCES_URL_PREFIX + API_KEY_URL_PARAM + apiKey;
+        post(url, new JSONObject().put(DATA_SOURCE_ID, dataSourceId).toString(), CheckResponseStatus.NO);
+        return true;
+    }
+
+    /**
+     * Removes user from user-group which are both specified by their ids in methods arguments.
+     *
+     * @return false if user is not a member of group, true if it is and then it will be successfully removed.
+     * @throws IllegalArgumentException if either user or user-group does not exist.
+     * @throws IOException              if server is not available due to connection error or API key is invalid.
+     */
+    public boolean removeUserFromGroup(int userId, int groupId) throws IOException {
+        checkIfEntityExists(User.class, userId);
+        UserGroup userGroup = getWithUsersAndDataSources(groupId);
+        if (userGroup.getUsers().stream().noneMatch(u -> u.getId() == userId)) {
+            return false;
+        }
+        String url = baseUrl + GROUPS_URL_PREFIX + "/" + groupId + MEMBERS_URL_PREFIX + "/" + userId + API_KEY_URL_PARAM + apiKey;
+        delete(url, CheckResponseStatus.NO);
+        return true;
```

Comment:
Same as above - how is the state of the server after this command being asserted?